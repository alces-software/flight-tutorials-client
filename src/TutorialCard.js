/*=============================================================================
 * Copyright (C) 2017 Stephen F. Norledge and Alces Flight Ltd.
 *
 * This file is part of Flight Launch.
 *
 * All rights reserved, see LICENSE.txt.
 *===========================================================================*/
import React from 'react';

import Card from './Card';
import Markdown from 'react-markdown';
import Overlay from './TutorialCardOverlay';
import './styles/TutorialCard.scss';

const TutorialCard = ({
  onSelectTutorial,
  tutorial,
}) => (
  <Card
    className="TutorialCard"
    onClick={onSelectTutorial}
    subtitle={tutorial.subtitle}
    subtitleSize="medium"
    title={tutorial.title}
    titleLogoOnRight
    titleLogoUrl={tutorial.logoUrl}
    titleSize="large"
  >
    <Markdown source={tutorial.description} />
    <Overlay onSelectTutorial={onSelectTutorial} />
  </Card>
);

export default TutorialCard;
