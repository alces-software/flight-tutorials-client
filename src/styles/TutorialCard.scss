/*=============================================================================
 * Copyright (C) 2016 Stephen F. Norledge and Alces Flight Ltd.
 *
 * This file is part of Flight Tutorials.
 *
 * All rights reserved, see LICENSE.txt.
 *===========================================================================*/

$card-width: 560px;
$card-height: 425px;

.TutorialCard {
    width: $card-width;
    height: $card-height;
}
