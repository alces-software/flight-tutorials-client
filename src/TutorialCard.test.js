// @flow
/*=============================================================================
 * Copyright (C) 2017 Stephen F. Norledge and Alces Flight Ltd.
 *
 * This file is part of Flight Tutorials.
 *
 * All rights reserved, see LICENSE.txt.
 *===========================================================================*/
import React from 'react';
import { render } from 'react-dom';
import renderer from 'react-test-renderer';

import TutorialCard from './TutorialCard';

const tutorial = {
  title: 'Tutorial 1',
  description: 'Tutorial 1 description.\n\nWith *markdown* **support**.',
  firstStep: 'step1',
  steps: {
    step1: {
      title: 'Tutorial 1 step 1',
      description: 'Tutorial 1 step 1 description',
      matches: [],
    },
  },
};

function renderCard() {
  return (
    <TutorialCard
      onSelectTutorial={() => {}}
      tutorial={tutorial}
    />
  );
}

it('renders without crashing', () => {
  const node = document.createElement('div');
  render(renderCard(), node);
});

it('renders correctly', () => {
  const tree = renderer.create(
    renderCard(),
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

