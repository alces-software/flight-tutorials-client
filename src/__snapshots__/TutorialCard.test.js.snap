// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="card-wrapper"
>
  <div
    className="card TutorialCard card--logo-right"
    onClick={[Function]}
  >
    <div
      className="card-block"
    >
      <div
        className="card-title-block"
      >
        <div
          className="card-title-block-inner"
        >
          <div
            className="card-title-logo-container"
          >
            <div
              className="card-title-logo-helper"
            />
          </div>
          <h2
            className="card-title large"
            title="Tutorial 1"
          >
            Tutorial 1
          </h2>
        </div>
        <h6
          className="card-subtitle medium"
        />
      </div>
      <div>
        <p>
          Tutorial 1 description.
        </p>
        <p>
          With 
          <em>
            markdown
          </em>
           
          <strong>
            support
          </strong>
          .
        </p>
      </div>
      <div
        className="flight-cardOverlay"
      >
        <button
          className="btn btn-link"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          <span
            aria-hidden={true}
            className="fa fa-search-plus fa-2x flight-icon flight-icon-search-plus"
            onBlur={[Function]}
            onClick={null}
            onFocus={[Function]}
            onMouseOut={[Function]}
            onMouseOver={[Function]}
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
